app/assets/javascripts/components/PipelineSampleReport.jsx

```diff
@@ -63,19 +77,42 @@ class PipelineSampleReport extends React.Component {
     return params.selected_genus != 'None' && params.disable_filters != 1;
   }
 
+  //path to NCBI 
+  gotoNCBI(taxId) {
+    const ncbiLink = `https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=${taxId}`;
+    window.open(ncbiLink, '_blank');
+  }
+
+  //download Fasta  
+  downloadFastaUrl(reportDetails, taxInfo) {
+    location.href = `/pipeline_outputs/${reportDetails.pipeline_info.id}/fasta/${taxInfo.tax_level}/${taxInfo.tax_id}/NT_or_NR`;
+  }
+
+  displayTags(taxInfo, reportDetails) {
+    return (
+      <span className="link-tag">
+        <i onClick={this.gotoNCBI.bind(this, taxInfo.tax_id)} className="fa fa-link cloud" aria-hidden="true"></i>
```

Comment:
Will this create a closure for every row in the table?   That's too many closures, which will cause the browser to pause for long reports.  Find a way to extract the information differently, e.g. from the class name.

Reply:
Thanks.